src/include/txn.i

```diff
@@ -442,6 +442,45 @@ __wt_txn_op_apply_prepare_state(
 	WT_REF_SET_STATE(ref, previous_state);
 }
 
+/*
+ * __wt_txn_op_delete_commit_apply_timestamps --
+ *	Apply the correct start and durable timestamps to any
+ *	updates in the page del update list.
+ */
+static inline void
+__wt_txn_op_delete_commit_apply_timestamps(
+    WT_SESSION_IMPL *session, WT_REF *ref)
+{
+	WT_TXN *txn;
+	WT_UPDATE **updp;
+	wt_timestamp_t ts;
+	uint32_t previous_state;
+
+	txn = &session->txn;
+	ts = txn->commit_timestamp;
+
+	/*
+	 * Lock the ref to ensure we don't race with eviction freeing the page
+	 * deleted update list or with a page instantiate.
+	 */
+	for (;; __wt_yield()) {
+		previous_state = ref->state;
+		WT_ASSERT(session, previous_state != WT_REF_READING);
```

Comment:
Can leave a comment explaining why the prev state cant be reading

Reply:
I don't think we need a comment here hopefully it's clear enough that we shouldn't be running this function if our current page is being read in.